externalDependencies and lodash causes error

### Command

serve

### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

17.0.0

### Description

When using the application builder's ``externalDependencies`` property to make ``lodash`` external, ng serve responds with the following error the first time, lodash is used:

```
X [ERROR] The entry point "lodash" cannot be marked as external
```

This behavior does not occur with ``@angular/core``, ``@angular/common``, or any other dependency directly installed by the CLI.

### Minimal Reproduction

Reproduction:

1. Create a new Angular app (``ng new``)
2. ``npm i lodash``
3. ``npm i @types/lodash -D``
4. Add this option to the application builder in the ``angular.json``:             
    ```
    "externalDependencies": ["lodash"],
    ```
5. Use ``lodash``, e. g. in your ``AppComponent`` (see https://github.com/manfredsteyer/external-reproduction/blob/main/src/app/app.component.ts)
6. ``ng serve -o``
7. Once ``lodash`` was used for the first time, the above-mentioned error message is shown on the console

Please find a minimal example reproducing the issue here:
https://github.com/manfredsteyer/external-reproduction.git

### Exception or Error

```text
X [ERROR] The entry point "lodash" cannot be marked as external
```


### Your Environment

```text
Angular CLI: 17.0.7
Node: 20.10.0
Package Manager: npm 10.2.3
OS: win32 x64

Angular: 17.0.8
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1700.7
@angular-devkit/build-angular   17.0.7
@angular-devkit/core            17.0.7
@angular-devkit/schematics      17.0.7
@angular/cli                    17.0.7
@schematics/angular             17.0.7
rxjs                            7.8.1
typescript                      5.2.2
zone.js                         0.14.2
```


### Anything else relevant?

_No response_

--- a/packages/angular_devkit/build_angular/src/builders/application/execute-build.ts
+++ b/packages/angular_devkit/build_angular/src/builders/application/execute-build.ts
@@ -178,12 +178,21 @@ export async function executeBuild(
 
   // Analyze external imports if external options are enabled
   if (options.externalPackages || bundlingResult.externalConfiguration) {
-    const { browser = new Set(), server = new Set() } = bundlingResult.externalImports;
-    // TODO: Filter externalImports to generate third argument to support wildcard externalConfiguration values
+    const {
+      externalConfiguration,
+      externalImports: { browser, server },
+    } = bundlingResult;
+    const implicitBrowser = browser ? [...browser] : [];
+    const implicitServer = server ? [...server] : [];
+    // TODO: Implement wildcard externalConfiguration filtering
     executionResult.setExternalMetadata(
-      [...browser],
-      [...server],
-      bundlingResult.externalConfiguration,
+      externalConfiguration
+        ? implicitBrowser.filter((value) => !externalConfiguration.includes(value))
+        : implicitBrowser,
+      externalConfiguration
+        ? implicitServer.filter((value) => !externalConfiguration.includes(value))
+        : implicitServer,
+      externalConfiguration,
     );
   }
 

